internal/controllers/v1beta1/garbagecollection/garbagecollection.go

```diff
@@ -78,6 +80,10 @@ func (c *GarbageCollectionController) Reconcile(ctx context.Context, request rec
 	certIface := c.certmanagerClient.CertmanagerV1().Certificates(request.Namespace)
 	cert, err := certIface.Get(ctx, request.Name, metav1.GetOptions{})
 	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
```

Comment:
This fixes a race condition when there are more than 1 kanopy-cert-controller pod, both garbageCollection Reconcile gets called at the same time for a certificate. Pod 1 deletes the cert first, and pod 2 arrives at this err. Previously it would requeue indefinitely. Now it would just return normally.

Reply:
After talking about leader election, this isn't necessary, I'll remove